micras_proxy/src/button.cpp

```diff
@@ -52,10 +59,8 @@ bool Button::update_state() {
             this->is_debouncing = false;
         }
     } else {
-        return raw_reading;
+        this->current_state = raw_reading;
```

Comment:
```suggestion
        this->is_debouncing = false;
        this->previous_state = current_state;
        this->current_state = raw_reading;
```

Ta faltando setar a flag de debounce pra false de novo, e acho q se pa faz sentido mover o `this->previous_state = current_state;` pra ca, pq é a funcao de update_state

Reply:
Impossível que o botão do micras tava sem debounce todo esse tempo kkkkk

Reply:
Eu deixei a atualização fora dessa classe porque ela não usa esse valor, tipo, saber o estado anterior só tem propósito pra determinar o status do robô, então achei que fazia mais sentido deixar lá mesmo.